src/shared/actions/alerts.js

```diff
@@ -47,8 +47,8 @@ const dismiss = () => ({ type: AlertsActionTypes.HIDE });
  *
  * @returns {function} dispatch
  */
-export const generateAlert = (category, title, message, closeInterval, err) => (dispatch) => {
-    dispatch(generate(category, title, message, closeInterval));
+export const generateAlert = (category, title, message, closeInterval, err, timeout) => (dispatch) => {
+    setTimeout(() => dispatch(generate(category, title, message, closeInterval)), timeout);
```

Comment:
```suggestion
    if (!timeout) {
        dispatch(generate(category, title, message, closeInterval));
    } else {
        setTimeout(() => dispatch(generate(category, title, message, closeInterval)), timeout);
    }
```

If `generateAlert` is called and `timeout` is not specified, a timer will be created with a timeout of 0. On mobile, this actually results in a [call](https://github.com/facebook/react-native/blob/c20070f10458d48d6ac1eaac49e681e932bfb9fd/Libraries/Core/Timers/JSTimers.js#L259) to the native timing module. Even though the native timing module will immediately call the callback if `timeout` is 0 (see [iOS](https://github.com/facebook/react-native/blob/c20070f10458d48d6ac1eaac49e681e932bfb9fd/React/Modules/RCTTiming.m#L280-L284) and [Android](https://github.com/facebook/react-native/blob/c20070f10458d48d6ac1eaac49e681e932bfb9fd/ReactAndroid/src/main/java/com/facebook/react/modules/core/Timing.java#L347-L352)), using `setTimeout` for something that will run immediately is still unnecessary IMHO. This change will also resolve most of the failing tests for `shared`.

Reply:
Okay good catch.